src/main/java/ai/preferred/venom/Crawler.java

```diff
@@ -281,7 +281,9 @@ private void except(final Job job, final Throwable ex) {
       synchronized (pendingJobs) { // Synchronisation required to prevent crawler stopping incorrectly.
```

Comment:
Do we need a set of pending jobs? 'connections' lock provides the same information, doesn't it?

Reply:
No, it doesn't. Connection releases once there is a response. PendingJobs only releases once the parsing/exception handling is finished. 

I can convert it to a semaphore if you want since we don't need the job information.

Reply:
Changed to AtomicInteger